### PR TITLE
👌 IMPROVE: BroadcastFilter

### PR DESCRIPTION
**Pull request by chrisjsewell**
For use with https://github.com/aiidateam/plumpy/pull/212

### CODE REVIEW

File: kiwipy/filters.py [hunk 1]
```diff
@@ -22,15 +22,20 @@ def __name__(self):
         return 'BroadcastFilter'
 
     def __call__(self, communicator, body, sender=None, subject=None, correlation_id=None):
+        if self.is_filtered(sender, subject):
+            return None
+        return self._subscriber(communicator, body, sender, subject, correlation_id)
+
+    def is_filtered(self, sender, subject) -> bool:
         if subject is not None and self._subject_filters and \
                 not any([check(subject) for check in self._subject_filters]):
```
Comment (sphuber) [inline kiwipy/filters.py:31]:
```suggestion
                not any(check(subject) for check in self._subject_filters):
```

File: kiwipy/filters.py [hunk 2]
```diff
@@ -22,15 +22,20 @@ def __name__(self):
...
-            return
+            return True
 
         if sender is not None and self._sender_filters and \
                 not any([check(sender) for check in self._sender_filters]):
```
Comment (sphuber) [inline kiwipy/filters.py:35]:
```suggestion
                not any(check(sender) for check in self._sender_filters):
```

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @chrisjsewell . It looks like this is just a simple refactoring to factor out the `is_filtered` method, is it not? Or am I missing a change in functionality as well. Anyway, it would be good to include this in the commit message so it is a bit clearer what is happening and especially why.

Besides that the changes look good. I would just suggest to change the two list comprehensions to a generator. The intermediate comprehension is unnecessary.

**Comment by chrisjsewell** [comment]
Yep just a refactor cheers

**Review by sphuber — APPROVED** [review]

**Comment by sphuber** [comment]
I updated the commit title to say `REFACTORED` instead of `IMPROVE` and even tried to replace the icon with the recycle arrows you use, but apparently copy pasting from a commit that uses the emoji doesn't work....

**Pull request merged**